add details about running individual e2e test cases (#960)
vertica/vertica-kubernetes: DEVELOPER.md

Prerequisite steps:

1. Before you can run an individual test, you need to set up a communal endpoint. To set up a [MinIO](https://min.io/) endpoint, run the following make target:
>
> ```shell
> make setup-minio
> ```
>
To set up a different communal endpoint, see [Custom communal endpoints](#custom-communal-endpoints).

2. You also need to set up some environmental variables:
>
> ```shell
> export VERTICA_DEPLOYMENT_METHOD=vclusterops
> export LICENSE_FILE=/file_path/license.key
> export BASE_VERTICA_IMG=opentext/vertica-k8s:24.2.0-1
> export VERTICA_IMG=opentext/vertica-k8s:latest
> ```
>
If the total number of nodes used in a test case is more than 3, you have to set up LICENSE_FILE with the license file path.
If it is no more than 3, you must unset LICENSE_FILE:
>
> ```shell
> unset LICENSE_FILE
>
BASE_VERTICA_IMG and VERTICA_IMG are used for the upgrade test cases. The BASE_VERTICA_IMG is the base vertica vertion that will be installed. VERTICA_IMG is the vertica version that the base version will be upgraded to. The version in VERTICA_IMG must be higher than that in BASE_VERTICA_IMG.

> VERTICA_DEPLOYMENT_METHOD=vclusterops lets the backend use vcluster package to manage vertica clusters, instead of the legacy admintools.

Reviewer: Given we still support admintools, let's say by default its value is `admintools` and in that case the vertica image must be admintools compatible

   make run-int-tests | tee kuttl.out
   ```

### Run Individual e2e Tests

You can run individual tests from the command line with the [KUTTLE CLI](https://kuttl.dev/docs/cli.html#setup-the-kuttl-kubectl-plugin).

Prerequisite steps:

1. Before you can run an individual test, you need to set up a communal endpoint. To set up a [MinIO](https://min.io/) endpoint, run the following make target:
>
> ```shell
> make setup-minio
> ```
>
To set up a different communal endpoint, see [Custom communal endpoints](#custom-communal-endpoints).

2. You also need to set up some environmental variables:
>
> ```shell
> export VERTICA_DEPLOYMENT_METHOD=vclusterops
> export LICENSE_FILE=/file_path/license.key
> export BASE_VERTICA_IMG=opentext/vertica-k8s:24.2.0-1
> export VERTICA_IMG=opentext/vertica-k8s:latest
> ```
>
If the total number of nodes used in a test case is more than 3, you have to set up LICENSE_FILE with the license file path.
If it is no more than 3, you must unset LICENSE_FILE:
>
> ```shell
> unset LICENSE_FILE
>

> BASE_VERTICA_IMG and VERTICA_IMG are used for the upgrade test cases. The BASE_VERTICA_IMG is the base vertica vertion that will be installed. VERTICA_IMG is the vertica version that the base version will be upgraded to. The version in VERTICA_IMG must be higher than that in BASE_VERTICA_IMG.

Reviewer 2: Could you move the explanation of the environment variables right below the examples mentioned above?
For example:
```shell
export VERTICA_DEPLOYMENT_METHOD=vclusterops
VERTICA_DEPLOYMENT_METHOD=vclusterops lets the backend use ...
```shell
export LICENSE_FILE=/file_path/license.key
If the total number of nodes used ...
...

   ```

### Run Individual e2e Tests

You can run individual tests from the command line with the [KUTTLE CLI](https://kuttl.dev/docs/cli.html#setup-the-kuttl-kubectl-plugin).

Prerequisite steps:

1. Before you can run an individual test, you need to set up a communal endpoint. To set up a [MinIO](https://min.io/) endpoint, run the following make target:

```shell
make setup-minio
```

> To set up a different communal endpoint, see [Custom communal endpoints](#custom-communal-endpoints).

2. You also need to set up some environmental variables:
```shell
export VERTICA_DEPLOYMENT_METHOD=vclusterops
```
> VERTICA_DEPLOYMENT_METHOD=vclusterops lets the backend use vcluster package to manage vertica clusters. If it is not set, the default value will be admintools and the vertica image must be admintools compatible.

```shell
export LICENSE_FILE=/file_path/license.key
unset LICENSE_FILE
```
> If the total number of nodes used in a test case is more than 3, you have to set up LICENSE_FILE with the license file path.
If it is no more than 3, you must unset LICENSE_FILE.

```shell
export BASE_VERTICA_IMG=opentext/vertica-k8s:24.2.0-1
export VERTICA_IMG=opentext/vertica-k8s:latest
```

> > BASE_VERTICA_IMG and VERTICA_IMG are used for the upgrade test cases. The BASE_VERTICA_IMG is the base vertica vertion that will be installed. VERTICA_IMG is the vertica version that the base version will be upgraded to. The version in VERTICA_IMG must be higher than that in BASE_VERTICA_IMG.

Reviewer: ```suggestion
> VERTICA_IMG is the vertica image you want to run the test with. For upgrade test cases, BASE_VERTICA_IMG is the base vertica version that will be installed. VERTICA_IMG is the vertica version that the base version will be upgraded to. The version in VERTICA_IMG must be higher than that in BASE_VERTICA_IMG.
```

   make run-int-tests | tee kuttl.out
   ```

### Run Individual e2e Tests

You can run individual tests from the command line with the [KUTTLE CLI](https://kuttl.dev/docs/cli.html#setup-the-kuttl-kubectl-plugin).

Prerequisite steps:

1. Before you can run an individual test, you need to set up a communal endpoint. To set up a [MinIO](https://min.io/) endpoint, run the following make target:
>
> ```shell
> make setup-minio
> ```
>

>     To set up a different communal endpoint, see [Custom communal endpoints](#custom-communal-endpoints).

Reviewer 2: The explanation is currently inside a shell frame. We need to adjust the format. Could you try to remove the symbol ">" from lines 361 to 365? If this works, do the same thing for other env vars below.

Author: I made the change. It turns out ">" is used to add an indent to all lines in a graph.

    If it is no more than 3, you must unset LICENSE_FILE.

>
> ```shell
> export BASE_VERTICA_IMG=opentext/vertica-k8s:24.2.0-1
> export VERTICA_IMG=opentext/vertica-k8s:latest
> ```
>
    BASE_VERTICA_IMG and VERTICA_IMG are used for the upgrade test cases. The BASE_VERTICA_IMG is the base vertica vertion that will be installed. VERTICA_IMG is the vertica version that the base version will be upgraded to. The version in VERTICA_IMG must be higher than that in BASE_VERTICA_IMG.

3. kuttl-test.yaml is the configuration file for e2e test cases. There is a "timeout" field in it. If your server is not fast enough, you may need to increase that value to pass the test cases. There is another field "parallel" that controls the maximum number of tests to run at once. It is set to 2 by default. You can set it to 1 if your server is not fast enough.

4. To avoid downloading the same image multiple times, you can run the following commands to download the images and push them to the kind cluster before you run the test cases.
>
> ```shell
> scripts/push-to-kind.sh -i $VERTICA_IMG
> scripts/push-to-kind.sh -i $BASE_VERTICA_IMG
>

> This will speed up test case execution and avoid timeout.

Reviewer 2: This line needs indent as well.

Reviewer 2: This line needs to be fixed. It's still in a shell frame.

Author: It is fixed by the following commit
